executor/write.go

```diff
@@ -867,12 +881,16 @@ func (e *InsertExec) exec(ctx context.Context, rows [][]types.Datum) (Row, error
 }
 
 type keyWithDupError struct {
```

Comment:
Can we make this type contain one record key and multiple unique keys?
So it can represent all the duplicate keys in a row, we don't need a slice of it.
```go
type dupKeysInRow struct {
	recordKey       kv.Key
	recordKeyDupErr error
	uniqKeys        []kv.Key
	uniqKeysDupErrs []error
	newRowValue     []byte
}
```

Reply:
There is no special advantage in changing this. There are still many places to change, and the original logic of `insert ignore` will be more complicated. I am still more inclined to use a bool to distinguish the key is a record or an index. The unique key and errors are also separated, and it is also easy to cause an index out of range problem after bad use.